mmv1/third_party/terraform/services/sql/resource_sql_database_instance_test.go.tmpl

```diff
@@ -1183,6 +1183,53 @@ func TestAccSqlDatabaseInstance_withPSCEnabled_withIpV4Enabled(t *testing.T) {
 	})
 }
 
+func TestAccSqlDatabaseInstance_withPscEnabled_withNetworkAttachmentUri_thenRemoveNetworkAttachment(t *testing.T) {
+	t.Parallel()
+
+	instanceName := "tf-test-" + acctest.RandString(t, 10)
+	projectId := envvar.GetTestProjectFromEnv()
+	region := "us-central1"
+	networkName := acctest.BootstrapSharedTestNetwork(t, "tf-test-cloud-sql-network")
+	subnetworkName := acctest.BootstrapSubnet(t, "tf-test-cloud-sql-subnetwork", networkName)
+	networkAttachmentName := acctest.BootstrapNetworkAttachment(t, "tf-test-cloud-sql-update-na", subnetworkName)
+	networkAttachmentUri := fmt.Sprintf("projects/%s/regions/%s/networkAttachments/%s", projectId, region, networkAttachmentName)
+
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccSqlDatabaseInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSqlDatabaseInstance_withPSCEnabled_withoutPscOutbound(instanceName),
```

Comment:
Are we able to create the resource w/ the setting, i.e. could start with `testAccSqlDatabaseInstance_withPSCEnabled_withNetworkAttachmentUri` directly? No need to actually do so if it works- the update case is more interesting and we're not getting enough new information to justify the carrying cost of the test, since it's an API responsibility- but if it _doesn't_ we may want special handling & a dedicated test to confirm that it works as intended.

Reply:
We could optionally bridge over this with a `post_create` in Terraform and call the update immediately after creation when the value is specified. That would make recreation cases easier as you could do an all-in-one apply, provided it makes sense for the domain. The API supporting it and doing that serverside would be even better.

That's optional and non-blocking though; either way would be backwards compatible to add.

Reply:
This is correct! A post-create is how we'd support this in the provider unless there's a circular dependency in which case we'd need a different approach.